### ISSUE TITLE
LiveAlert not working on new Live page

### ISSUE THREAD

**Issue by unnecessarydave** [foxtrick-ng/foxtrick#18]
LiveAlerts do not show on the new match Live page.

It still works fine on Legacy Live.

**Comment by unnecessarydave** [comment]
I had a quick look at this.

The fundamentals of the alerts in live-alert.js are still functional.

Updating the CSS selectors:
23: `doc.querySelector('#ngLive .ht-tabs') :`
121: `ret = Foxtrick.toArray(tab.querySelectorAll('span.ellipsis'));`

Will allow Foxtrick to grab the correct doc nodes.

However, I am getting double alerts for each goal, one of which has the score 1 more than it should be, so more work needed.

**Comment by murko69** [comment]
Maybe you had a second tab open :) 
I remember the Foxtrick notifications for HT-Live somehow worked like that and gave me *2, sometimes *3 the notifications depending on how many HT-tabs I had open at the time.

**Comment by unnecessarydave** [comment]
I think I've found the reason for the double-notifications.  Foxtrick listens for changes to the Live scores banner up at the top and notifies when it detects that a score has changed.

However, I overlooked that it is also listening for changes to the main commentary section whereupon it calls runOverlay:
https://github.com/foxtrick-ng/foxtrick/blob/4907948188b046bfc9b345b62c805214f4217570/content/alert/live-alert.js#L45-L76
Notable here is that it gets the actual score from the Live scores banner, but then increments it by 1 for the team that scored.

I don't understand why we are listening for changes in two places.  At line 35 where the listener is registered there is a comment
https://github.com/foxtrick-ng/foxtrick/blob/4907948188b046bfc9b345b62c805214f4217570/content/alert/live-alert.js#L35-L38
but I don't know what it means by pre-announce.

I can of course simply disable runOverlay, but I'd be concerned that it is accounting for a situation that I am unaware of.

**Comment by unnecessarydave** [comment]
Well, disabling runOverlay certainly does the trick.  There was another fix needed so that it would alert for the first goal scored, but with that done it does seem to work.

I'll test it some more and then do a PR at some point.

**Comment by murko69** [comment]
Could the pre-announce part be about the stuff the commentator says before kick-off?
Simply meaning announcements pre-game?
That a sound and/or notification was created as soon as the commentator first shows?
That's ab out 60-75 seconds pre-game I think

**Comment by unnecessarydave** [comment]
Well that would make sense.   That's why I skipped over that part of the code initially, as I assumed it had nothing to do with goal notifications.  But, goal notifications is all runOverlay does.

runOverlay was added in Oct 2016 5568a3ab1a7f0b9839668a99f5e395a2018d5a0a and the commit message says 'notifies about goals the moment when they happen.'

So, I have a new theory: is it possible that back then the score tabs up at the top were slow to update?  That is, if you were in the match view in Live you would see the commentator announce a goal scored, goal scorer picture appears etc, but the score tab wouldn't update until a few moments later?

If that were the case then the code would make sense as written, and there would only ever be a single notification for the goal.
